openfe/protocols/openmm_rfe/equil_rfe_settings.py

```diff
@@ -107,11 +103,35 @@ class Config:
     alchemical water can be chosen. Default 0.8 * unit.nanometer.
     """
 
+    @validator('softcore_LJ')
+    def allowed_solvent(cls, v):
+        allowed_models = ['gapsys', 'beutler']
+        if v.lower() not in allowed_models:
+            errmsg = (
+                f"Only {allowed_models} softcore potentials are allowed, "
+                f"got {v.lower()}."
+            )
+            raise ValueError(errmsg)
+        return v
 
```

Comment:
```
In [3]: from typing import Literal

In [4]: class Thing(BaseModel):
   ...:     a: Literal['yes', 'no']
   ...: 

In [5]: Thing(a='yes')
Out[5]: Thing(a='yes')

In [6]: Thing(a='chicken')
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[6], line 1
----> 1 Thing(a='chicken')

File ~/miniconda3/envs/openfe_dev/lib/python3.11/site-packages/pydantic/main.py:341, in pydantic.main.BaseModel.__init__()

ValidationError: 1 validation error for Thing
a
  unexpected value; permitted: 'yes', 'no' (type=value_error.const; given=chicken; permitted=('yes', 'no'))
```

it looks like pydantic is smart enough to create this validator automatically when you specify a Literal.

But, it won't (easily) allow (partially) upper case versions of the literals, so 'Gapsys' here would annoyingly fail.

Not to be fixed here, but probably a quality of life thing we can fix in a future issue.

Reply:
Ok, will remove this validator then!